### PR TITLE
Fall back to using OIDC Subject instead of Email

### PR DESCRIPTION
**Pull request by aigarius**
Email is not mandatory field, Subject is mandatory and expected to be unique. Might want to take a look at UserInfo first, however.

Issue: #56

### CODE REVIEW

**Comment by aigarius** [comment]
This also needs a patch for the case of refresh ....

File: providers/oidc.go [hunk 1]
```diff
@@ -114,7 +115,8 @@ func (p *OIDCProvider) createSessionState(ctx context.Context, token *oauth2.Tok
 	}
 
 	if claims.Email == "" {
-		return nil, fmt.Errorf("id_token did not contain an email")
+		// TODO: Try getting email from /userinfo before falling back to Subject
+		claims.Email = claims.Subject
```
Comment (JoelSpeed) [inline providers/oidc.go]: If `claims.Email == ""` then would `claims.Verified` necessarily be unset? We might want to make a `user := claims.Subject` variable and then overwrite it with `claims.Email` if it is set and then we can check for `claims.Email != ""` before `claims.Verified != nil`, WDYT?

Comment (aigarius) [reply providers/oidc.go]: The verified takes it value from "email_verified" field, I would assume that if the backend does not provide use with an email, it also will not be saying whether email is verified. As I read the code the intent was to ignore the email if it was unverified. Maybe there are some backends out there that do send email_verified even without email. Don't know.

I'd rather do minimum possible right now and leave that handling to whoever implements getting email with separate request.

**Comment by aigarius** [comment]
> We will need to add a note to the changelog about this change

Not super sure about that. This will make this work on systems where it never worked before, but in a peculiar way. Once it is fully fixed correctly (as in email is requested separately from userinfo it is not present in the immediate reply) then the responses will change for these systems, but then also those systems will start working exactly the same as the systems that already work now.

**Comment by aigarius** [comment]
> We will need to add a note to the changelog about this change

Changelog added

**Pull request merged**